### DISCUSSION TITLE
Feature Request: Add To Swift Package Manager And Convert to Swift Code

### DISCUSSION THREAD

**Discussion by chitgoks** [ngageoint/simple-features-geojson-ios#2] [Q&A] [answered]
Feature Request: Add To Swift Package Manager and convert to Swift code from Objective C
Any plans?

**Comment by bosborn** [comment]
You can currently [use from Swift](https://github.com/ngageoint/simple-features-geojson-ios#swift) (corresponding [test](https://github.com/ngageoint/simple-features-geojson-ios/blob/master/sf-geojson-iosTests/SFGSwiftReadmeTest.swift)).  No plans to port the code itself to Swift but that shouldn't matter for dependent code.  An example would be our [Grid Swift library](https://github.com/ngageoint/grid-ios) which uses our [Simple Features Objective-C library](https://github.com/ngageoint/simple-features-ios).

Not opposed to Swift Package Manager support.  We use CocoaPods so hasn't been a priority.  I was messing with it a little bit on our TIFF library [here](https://github.com/ngageoint/tiff-ios/discussions/13), originating from [this pull request](https://github.com/ngageoint/tiff-ios/pull/12), bet never followed through.

**Comment by chitgoks** [comment]
@bosborn i got it to load. I have a little question though. i am not sure why it doesnt work the same way in android. i did it like this
```
FeatureCollection featureCollection = FeatureConverter.toFeatureCollection(output);
            for (Feature feature : featureCollection) {
                double[] bbox = feature.getBbox();
....
```

but in swift i cant do it like this
```
if let featureCollection = SFGFeatureConverter.json(toFeatureCollection: jsonString) {
                for feature in featureCollection {
                    let bbox = feature.bbox
```

says For-in loop requires 'SFGFeatureCollection' to conform to 'Sequence'.  it works ok if i do featureCollection.features

but then feature has no property bbox unlike in the Android library. what is the equivalent for the ios version?

or how to cast feature to any SFGPoint, SFGLineString, SFGPolygon, etc. if i do just featurethe tooltip says self: Any and not SFGFeature. but print(feature) does show it as SFGFeature.

thoughts?

**Comment by chitgoks** [comment]
Edit: I got it to work. Though i am confused why it had to be like that.
if i do it like this, it works
```
for feature in featureCollection.features {
                if let sfgFeature: SFGFeature = feature as? SFGFeature,
                   let point = sfgFeature.geometry as? SFGPoint {
```

weird that i have to specify the type in sfgFeature else it errors out. Maybe it is because this is objective c code? sorry, still new to ios.

**Comment by bosborn** [comment, marked as answer]
Yeah that works.  Objective-c code doesn't conform to the Swift Sequence protocol.  Looping over the SFGeometry typed NSMutableArray requires that extra type information.

Two alternates:

```
for feature in featureCollection.features where feature is SFGFeature {
    let bbox = (feature as! SFGFeature).bbox
}
```

```
for feature in featureCollection.features {
    if let sfgFeature = feature as? SFGFeature {
        let bbox = sfgFeature.bbox
    }
}
```
